internal/resources/provisioner/data_source_provisioner.go

```diff
@@ -55,18 +55,34 @@ func dataSourceProvisionerRead(ctx context.Context, d *schema.ResourceData, m in
 		return diag.FromErr(errors.Wrapf(err, "Couldn't read Tanzu Mission Control provisioner configurations."))
 	}
 
-	resp, err := config.TMCConnection.ProvisionerResourceService.ProvisionerResourceServiceList(model.FullName)
-	if err != nil {
-		if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
-			_ = schema.RemoveFromState(d, m)
-			return
+	if model.FullName.Name == "" {
+		resp, err := config.TMCConnection.ProvisionerResourceService.ProvisionerResourceServiceList(model.FullName)
+		if err != nil {
+			if clienterrors.IsNotFoundError(err) && !helper.IsDataRead(ctx) {
+				_ = schema.RemoveFromState(d, m)
+				return
+			}
+		}
+
+		for i := range resp.Provisioners {
+			d.SetId(resp.Provisioners[i].Meta.UID)
```

Comment:
If the array contains more than one response will it not overwrite the ID?

Reply:
The expectation is to set unique ID for every provisioner in the list